pkg/kubelet/kubelet.go

```diff
@@ -2296,9 +2296,7 @@ func (kl *Kubelet) canAdmitPod(pods []*v1.Pod, pod *v1.Pod) (bool, string, strin
 		// Use allocated resources values from checkpoint store (source of truth) to determine fit
 		otherPods := make([]*v1.Pod, 0, len(pods))
 		for _, p := range pods {
-			op := p.DeepCopy()
-			kl.updateContainerResourceAllocation(op)
-
+			op, _ := kl.statusManager.UpdatePodFromAllocation(p)
```

Comment:
hmmm....could we add a comment in `canAdmitPod` that it assumes the pod spec contains the allocated resources? Right now the `pods` argument takes the vanilla pod specs, while `pod` takes the spec with allocated resources. 

Another option is to change the arg name to reflect this:
```
canAdmitPod(pods []*v1.Pod, allocatedPod *v1.Pod)
```

And I know this is not the ideal state that we'd like, but making it more readable would help for now.

Reply:
I actually went the other direction, and moved the updating of `otherPods` out of this function, since now we can just use the `getAllocatedPods()` function, and renamed the pods argument accordingly.

The `pod` argument is actually usually _not_ the allocated pod, since we want to check the desired resources against admission to see if they can be allocated. Basically, there are 3 ways this function can be called:
1. New pod scheduled to this node: no allocation yet, pod is the desired pod
2. Resized pod: pod has an allocation, but we want to see if we can admit the new desired state
3. Kubelet restart, re-admitting existing pods: pod has an allocation; even if there's a pending resize, we want to re-admit the allocated state first to make sure we don't reject a pod that's already running. This is the only case where this function is called with an allocated pod.

Reply:
Enumerating the cases where the function is called is actually very helpful. Verifying my understanding below:
 - (1) and (3). Called in `HandlePodAddition` -- we look at the pod and its allocated resources (if it exists) and the desired spec (if it's considered new).
 - (2). Called in `canResizePod` -- we look at the pod's updated desired spec

Does that sound right to you?

Reply:
Yes, correct.

I'd add this as a comment, but I'm not sure where. I don't really like add comments about callers to functions, because in my experience they tend to be hard to keep up-to-date.

Reply:
Yeah not saying we need to add this to the function itself. I do think adding some comments at the call sites would help. Not a blocker though.